scripts/o3de/o3de/android.py

```diff
@@ -0,0 +1,598 @@
+    android_configure_subparser.add_argument('-p', '--project', type=str, required=False,
+                                             help="The name of the registered project to configure the settings for. This value is ignored if '--global' was specified."
+                                                  "Note: If both '--global' and '-p/--project' is not specified, the script will attempt to deduce the project from the "
+                                                  "current working directory if possible")
+    android_configure_subparser.add_argument('-l', '--list', default=False, action='store_true',
+                                             help='Display the current android settings. ')
+    android_configure_subparser.add_argument('--validate', default=False, action='store_true',
+                                             help='Validate the settings and values in the android settings. ')
+    android_configure_subparser.add_argument('--set-value', type=str, required=False,
+                                             help='Set the value for an android setting, using the format <argument>=<value>. For example: \'ndk_version=22.5*\'',
+                                             metavar='VALUE')
+    android_configure_subparser.add_argument('--clear-value', type=str, required=False,
+                                             help='Clear a previously configured setting.',
+                                             metavar='VALUE')
+    android_configure_subparser.add_argument('--set-password', type=str, required=False,
+                                             help='Set the password for a password setting. A password prompt will be presented.',
+                                             metavar='SETTING')
+    android_configure_subparser.add_argument('--debug',
+                                             help=f"Enable debug level logging for this script.",
+                                             action='store_true')
+    android_configure_subparser.set_defaults(func=configure_android_options)
+
+
+    #
+    # Configure the subparser for 'android_generate'
+    #
+    android_generate_subparser = subparsers.add_parser(O3DE_COMMAND_GENERATE,
+                                                        help='Generate an android/gradle project.')
+
+    # Project Name
+    if project_name:
+        android_generate_subparser.add_argument('-p', '--project', type=str, required=False,
+                                                 help="The name of the registered project to generate the android project for. If not supplied, the current detected project from the "
+                                                      f"current path ({project_name}) will be used.")
+    else:
+        android_generate_subparser.add_argument('-p', '--project', type=str, required=True,
+                                                help="The name of the registered project to generate the android project for.")
```

Comment:
I think it is going to be error prone to set the optionality of arguments based on if an android config file has been read.

Reply:
Thats a good point, but I would still like to retain the functionality of attempting to resolve the project based on the CWD. I'll remove the required flag and handle this this use case better